oauthproxy_test.go

```diff
@@ -1360,17 +1461,21 @@ type SignatureAuthenticator struct {
 }
 
 func (v *SignatureAuthenticator) Authenticate(w http.ResponseWriter, r *http.Request) {
+	var err error
 	result, headerSig, computedSig := v.auth.AuthenticateRequest(r)
 	if result == hmacauth.ResultNoSignature {
-		w.Write([]byte("no signature received"))
+		_, err = w.Write([]byte("no signature received"))
 	} else if result == hmacauth.ResultMatch {
-		w.Write([]byte("signatures match"))
+		_, err = w.Write([]byte("signatures match"))
 	} else if result == hmacauth.ResultMismatch {
-		w.Write([]byte("signatures do not match:" +
+		_, err = w.Write([]byte("signatures do not match:" +
 			"\n  received: " + headerSig +
 			"\n  computed: " + computedSig))
 	} else {
-		panic("Unknown result value: " + result.String())
+		panic("unknown result value: " + result.String())
+	}
+	if err != nil {
+		panic(err)
 	}
```

Comment:
could make this a switch that sets the value of a string, then handle the error writing afterwards, WDYT?

Reply:
I'll give it a quick whirl! But I think this will get a much bigger overhaul when I tackle all this with this issue: https://github.com/oauth2-proxy/oauth2-proxy/issues/648

Thinking back, I feel like a few PRs in the past have added headers to the SignatureHeaders yet didn't mention that this would be a breaking change requiring any users of this functionality to update how the backend code validated the signature.

So I'm really starting to think no one uses this -- or any few users are very advanced and know to check the code itself to see what headers they need to hmacauth on the backend (since we don't document it).